test/spec/directives/document-line-edit.js

```diff
@@ -719,6 +719,106 @@ describe('Directive: documentLineEdit', function () {
+        beforeEach(function () {
+          scope.line = {name: 'image', value: 'foo:bar', warnings: ['broke', 'wrong']};
+          element = compile('<document-line-edit line="line"></document-line-edit>')(scope);
+          scope.$digest();
+        });
+
+        it('joins the warnings together', function() {
+          expect(element.isolateScope().warningMessages()).toEqual('broke<br/>wrong');
+        });
+      });
+
+      describe('when a line has neither', function() {
+        beforeEach(function () {
+          scope.line = {name: 'image', value: 'foo:bar'};
+          element = compile('<document-line-edit line="line"></document-line-edit>')(scope);
+          scope.$digest();
+        });
+
+        it('joins the warnings together', function() {
+          expect(element.isolateScope().warningMessages()).toEqual('');
+        });
+      });
+
+      describe('when a line has both', function() {
+        beforeEach(function () {
+          scope.line = {name: 'image', value: 'foo:bar', warning: ['messed', 'up'], subWarnings: {1: ['busted', 'wrong']}};
+          element = compile('<document-line-edit line="line"></document-line-edit>')(scope);
+          scope.$digest();
+        });
+
+        it('joins the subwarnings together and ignores the top level warnings', function() {
+          expect(element.isolateScope().warningMessages(0)).toEqual('busted<br/>wrong');
+        });
+      });
+    });
+
+    describe('scope.errorMessages', function () {
+      describe('when a line has subErrors', function() {
+        beforeEach(function () {
+          scope.line = {name: 'image', value: 'foo:bar', subErrors: {1: ['busted', 'wrong']}};
+          element = compile('<document-line-edit line="line"></document-line-edit>')(scope);
+          scope.$digest();
+        });
+
+        it('joins the errors together', function() {
+          expect(element.isolateScope().errorMessages(0)).toEqual('busted<br/>wrong');
+        });
+      });
+
+      describe('when a line has errors', function() {
+        beforeEach(function () {
+          scope.line = {name: 'image', value: 'foo:bar', errors: ['broke', 'wrong']};
+          element = compile('<document-line-edit line="line"></document-line-edit>')(scope);
+          scope.$digest();
+        });
+
+        it('joins the errors together', function() {
+          expect(element.isolateScope().errorMessages()).toEqual('broke<br/>wrong');
+        });
+      });
+
+      describe('when a line has neither', function() {
+        beforeEach(function () {
+          scope.line = {name: 'image', value: 'foo:bar'};
+          element = compile('<document-line-edit line="line"></document-line-edit>')(scope);
+          scope.$digest();
+        });
+
+        it('joins the errors together', function() {
+          expect(element.isolateScope().errorMessages()).toEqual('');
+        });
+      });
+
+      describe('when a line has both', function() {
```

Comment:
hey @dharmamike - I wanna make sure and go over this logic with you... I took an educated guess...